ParkingLotException File

Sutantu415 opened the pull request:
Made an exception class/file to handle all exceptions for the parkingLot class

Sutantu415 committed:
ParkingLotException File

Made an exception class/file to handle all exceptions for the parkingLot class

Sutantu415 commented:
As mentioned in the messages it only has an exception message method currently. I'm thinking about how to handle illegal values in the file itself which I think is the end goal that we discussed on Saturday

PrancingGazelle commented on ParkingLot.java:
```diff
@@ -30,14 +30,14 @@ public ParkingLot(int parkingLotSize) {
      *@param parkingLotSpot is the parking spot that you are getting the status on
      *@return returns false if the slot is full/car is parked there or true for empty/car is not parked there
      **/
     public boolean getStatus(int parkingLotSpot) throws Exception {
```
since you've created a specific exception, you should update this "throws" to specify that exception type.

PrancingGazelle commented on ParkingLot.java:
```diff
@@ -55,12 +55,12 @@ public int getTotalParkingSpaces() {
+     *indicating you have parked successfully, else throws exception for an invalid value
      *@param parkingSpotToPark is the parking spot that they are trying to park at
      **/
     public void park(int parkingSpotToPark) throws Exception {
```
same comment as above. use throws to specify the particular type of exception thrown. later if you have more than one exception thrown, you can say "throws <exceptiontype1>, <exceptiontype2>..."

PrancingGazelle commented on ParkingLotException.java:
```diff
@@ -0,0 +1,5 @@
+public class ParkingLotException extends Exception {
```
this is a good start, but right now it is not very useful since it takes just a string and puts it into message. Think about how to pass back information about the exceptions as fields that are relevant to the parking lot library.

Sutantu415 replied:
There's a second commit in this pull request where I passed the value the user enters as a parameter to the exception method and added a field to the Exception class that will hold the illegal value to pass that information back to the user.

PrancingGazelle reviewed (changes requested):
good start...take a look at my feedback for improvements.

Sutantu415 committed:
ParkingLotException Update

I updated the parking lot exception class to store a field that has information about the illegal value causing the exception

Sutantu415 committed:
Extraction Method

Wrote an extraction method that does both parking and un-parking based on user input and changed the park and unpark method to the extraction method.

Sutantu415 committed:
Added Test Cases

Added a Test Case java file with 6 test cases to make sure the code's function works as its supposed to.
